lib/services/local/desktops.go

```diff
@@ -40,6 +40,25 @@ func NewWindowsDesktopService(backend backend.Backend) *WindowsDesktopService {
 
 // GetWindowsDesktops returns all Windows desktops matching filter.
 func (s *WindowsDesktopService) GetWindowsDesktops(ctx context.Context, filter types.WindowsDesktopFilter) ([]types.WindowsDesktop, error) {
+	// do a point-read instead of a range-read if a filter is provided
+	if filter.HostID != "" && filter.Name == "" {
```

Comment:
Should we have the same for `ListWindowsDesktops`?

Should we also optimize getting desktops served by a particular host ID?

Reply:
Yes and yes!